2-Draft-Accepted/RFCNNN-HelpSystem-As-A-Module.md

```diff
@@ -0,0 +1,62 @@
+---
+RFC: RFC<four digit unique incrementing number assigned by Committee, this shall be left blank by the author>
+Author: Aditya Patwardhan
+Status: Draft
+SupercededBy: <link to another RFC>
+Version: 0.2
+Area: Help System
+Comments Due: 03/30/2018
+Plan to implement: Yes
+---
+
+# Help System as a standalone PowerShell Module  
+
+This RFC proposes to remove the Help System from System.Management.Automation.dll and make it a standalone PowerShell module.
+
+## Motivation
+
+Currently, help system is part of System.Management.Automation.dll.
+Help system is used predominantly in interactive sessions and hence not required for automation scenarios.
+Since, it is part of System.Management.Automation.dll it is always loaded with PowerShell.
+Help system is sizeable piece of code, which increases the runtime footprint of the process.
+This RFC proposes to make the help system a PowerShell module, so it is loaded only when used.
+
+## Specification
+
+All the help system code resides under [help folder](https://github.com/PowerShell/PowerShell/tree/master/src/System.Management.Automation/help).
+This code will be moved to a separate assembly and be published as a module.
+
+All the public APIs from the help system will be moved to the new namespace 'Microsoft.PowerShell.HelpSystem'.
+This will be breaking change for code taking dependency of these public APIs.
+
+| Public API | Breaking change impact assessment
+| ------------ | --------
+| HelpCatergory | Currently internal, will be changed to public.
+| HelpCategoryInvalidException | Exception class, breaking change impact is low.
+| GetHelpCommand | Sealed class, breaking change impact is low.
+| GetHelpCodeMethods | Used by formatting and output, refactor to remove Help System dependency.
+| HelpNotFoundException | Exception class, breaking change impact is low.
+| SaveHelpCommand | Sealed class, breaking change impact is low.
+| UpdatableHelpCommandBase | Used by UpdateHelpCommand, breaking change impact is medium.
+| UpdateHelpCommand | Sealed class, breaking change impact is low.
+
+[CLR Type Forwarding](https://docs.microsoft.com/en-us/dotnet/framework/app-domains/type-forwarding-in-the-common-language-runtime) will be considered to lessen the impact of the breaking change.
+
+The interactive user experience would not change as command discovery will find the ```Get-Help```, ```Save-Help``` and ```Update-Help``` commands from the 'Microsoft.PowerShell.HelpSystem' module instead from 'Microsoft.PowerShell.Core' module (System.Management.Automation.dll).
```

Comment:
We need to make sure that scripting scenarios with fully qualified module names (Microsoft.PowerShell.Core\Get-Help) is **NOT** broken. PowerShell engine should support both - old module qualified names and the new module qualified names.